src/sparse_tabular.jl

```diff
@@ -249,19 +255,16 @@ function observation_matrix_a_sp_o(pomdp::POMDP)
             end
         end
     end
-    obsmats_A_SP_O = [sparse(obsmat_row_A[a], obsmat_col_A[a], obsmat_data_A[a], n_states(pomdp), n_states(pomdp)) for a in 1:n_actions(pomdp)]
-    @assert all(all(sum(obsmats_A_SP_O[a], dims=2) .≈ ones(n_observations(pomdp))) for a in 1:n_actions(pomdp)) "Observation probabilities must sum to 1"
+    obsmats_A_SP_O = [sparse(obsmat_row_A[a], obsmat_col_A[a], obsmat_data_A[a], ns, ns) for a in 1:na]
+    @assert all(all(sum(obsmats_A_SP_O[a], dims=2) .≈ ones(no)) for a in 1:na) "Observation probabilities must sum to 1"
     return obsmats_A_SP_O
 end
 
 # MDP and POMDP common methods
 
-POMDPs.n_states(prob::SparseTabularProblem) = size(prob.T[1], 1)
-POMDPs.n_actions(prob::SparseTabularProblem) = size(prob.T, 1)
-
-POMDPs.states(p::SparseTabularProblem) = 1:n_states(p)
-POMDPs.actions(p::SparseTabularProblem) = 1:n_actions(p)
-POMDPs.actions(p::SparseTabularProblem, s::Int64) = [a for a in actions(p) if sum(transition_matrix(p, a)) ≈ n_states(p)]
+POMDPs.states(p::SparseTabularProblem) = 1:size(p.T[1], 1)
```

Comment:
Should we call `collect` on these?

Reply:
As is, `states` is an iterator, one can call `collect(states(mdp))`. I don't think we want to `collect` by default, what would be the use case?

Reply:
I don't think we should collect by default